protos/organization.proto

```diff
@@ -0,0 +1,70 @@
+// Copyright 2018 Bitwise IO
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// -----------------------------------------------------------------------------
+
+syntax = "proto3";
+
+message Organization {
+    message Holding {
+        enum AssetType {
+            CURRENCY = 0;
+            BOND = 1;
+        }
+        // Whether the asset is currency or a bond
+        AssetType asset_type = 1;
+
+        // If asset_type is BOND, the asset_id will contain the bond_id
+        // If asset_type is CURRENCY, this will be the string ‘USD’
+        string asset_id = 2;
+
+        // The current balance
+        float amount = 3;
+    }
+
+    enum OrganizationType {
+        TRADING_FIRM = 0;
+        PRICING_SOURCE = 1;
+    }
+
+    enum Role {
+        MARKETMAKER = 0;
+        TRADER = 1;
+    }
+
+    // Ticker symbol associated with the organization
+    string organization_id = 1;
+
+    // Name of the organization
+    string name = 2;
+
+    // Type of the organization
+    OrganizationType organization_type = 3;
+
+    // Industry of the organization
+    string industry = 4;
+
+    // List of the assets that the organization owns
+    repeated Holding holdings = 5;
+
+    // Four-letter code representing the identity of the organization for
+    // the purpose of providing quotes to the market
+    string pricing_source = 6;
+
+    // Map of participants to roles
+    map<string, Role> authorization = 7;
```

Comment:
Would it make sense to have authorization be similar to holdings, as in a repeated field? I think it might increase clarity.

Reply:
That sounds like a good idea. I'll update it